lib/alertmanager_api/silence.py

```diff
@@ -0,0 +1,198 @@
+import logging
+import requests
+from exceptions.missing_mandatory_param_error import MissingMandatoryParamError
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import RequestException
+from alertmanager_api.silence_details import SilenceDetailsHandler
+
+
+class AlertManagerAPI:
+    """
+    class implementing an interface to the Kayobe's alertmanager
+    functionalities:
+        * create a silence
+        * remove an existing silence
+    """
+
+    def __init__(self, alertmanager_account):
+        """
+        :param alertmanager_account: object with credentials
+        :type alertmanager_account: AlertManagerAccount
+        """
+        self.log = logging.getLogger("AlertManagerAPI")
+        self.alertmanager_account = alertmanager_account
+        self.auth = HTTPBasicAuth(
+            alertmanager_account.username, alertmanager_account.password
+        )
+        self.api_endpoint = alertmanager_account.alertmanager_endpoint
+
+    def schedule_silence(self, silence):
+        """
+        Schedules a silence in alertmanager
+
+        :param silence: object with the specs to create a silence
+        :type silence: SilenceDetails
+        :return: a new silence object, including the silence_id
+        :rtype: SilenceDetails
+        :raises MissingMandatoryParamError:
+            if any of the required attributes are missing
+            in the silence object:
+            instance_name
+            start time
+            end time
+            author
+            comment
+        :raises requests.exceptions.RequestException:
+            when the request to the AlertManager failed
+        """
+        if not silence.instance_name:
+            msg = "Missing silence instance name"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.start_time_dt:
+            msg = "Missing silence start time"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.end_time_dt:
+            msg = "Missing silence end time"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.author:
+            msg = "Missing silence author"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.comment:
+            msg = "Missing silence comment"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
```

Comment:
unnecessary - this function is doing too many things 

validating the values of `silence` dataclass should happen when the object is created - separation of concerns. See my comment on making a dataclass in `silence_details.py`